File: problemSolving/python/baekjoon/step/bruteforce/blackjack.py
N, M = map(int, input().split())
nums = list(map(int, input().split()))
answer = 0
for i in range(N):
  for j in range(i + 1, N):
    for k in range(j + 1, N):
      if nums[i] + nums[j] + nums[k] <= M:
        if M - answer > M - (nums[i] + nums[j] + nums[k]):
          answer = nums[i] + nums[j] + nums[k]
print(answer)

File: problemSolving/baekjoon/step/bruteforce/javascript/blackjack.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./input.txt")
  .toString()
  .trim()
  .split("\n")
  .map((el) => el.split(" ").map(Number));

const [N, M] = input[0];
const numbers = input[1];

function find(n, m, nums) {
  let answer = 0;
  let sum;

  for (let i = 0; i < N; i++) {
    for (let j = i + 1; j < N; j++) {
      for (let k = j + 1; k < N; k++) {
        sum = numbers[i] + numbers[j] + numbers[k];

        if (sum <= M && sum > answer) {
          answer = sum;
        }
      }
    }
  }
  return answer;
}

console.log(find(N, M, numbers));

